ee/modules/140-user-authz/images/webhook/web/hook/handler.go

```diff
@@ -313,14 +356,44 @@ func (h *Handler) affectedDirs(r *WebhookRequest) []DirectoryEntry {
 	return dirEntriesAffected
 }
 
+// getsthe map of namespaces matching specific label selector from k8s api client
+func (h *Handler) getNamespacesByLabelSelector(namespaceSelector NamespaceSelector) (map[string]struct{}, error) {
+	namespaces := make(map[string]struct{})
+
+	getNamespaces, err := h.kubeclient.CoreV1().Namespaces().List(context.TODO(), metav1.ListOptions{LabelSelector: metav1.FormatLabelSelector(namespaceSelector.LabelSelector)})
+	if err != nil {
+		return nil, err
+	}
+
+	for _, namespace := range getNamespaces.Items {
+		namespaces[namespace.Name] = struct{}{}
+	}
+	return namespaces, nil
+}
+
+// checks if a namespace name matches any system namespace regex
+func namespaceIsSystem(namespace string, systemNamespacesRegex []*regexp.Regexp) bool {
+	for _, pattern := range systemNamespacesRegex {
+		if pattern.MatchString(namespace) {
+			return true
+		}
+	}
+	return false
+}
+
 func hasLimitedNamespaces(entry *DirectoryEntry) bool {
-	if len(entry.LimitNamespaces) == 0 || entry.LimitNamespacesAbsent {
+	if (len(entry.LimitNamespaces) == 0 && len(entry.NamespaceSelectors) == 0) || entry.LimitNamespacesAbsent {
```

Comment:
NIT: to avoid nesting brackets:
```suggestion
	if len(entry.LimitNamespaces) + len(entry.NamespaceSelectors) == 0 || entry.LimitNamespacesAbsent {
```

Reply:
The right way will be to check namespace selectors in the other method.